tests/unit/bokeh/core/property/test_bases.py

```diff
@@ -212,6 +212,19 @@ def test_validation_on(self) -> None:
         bcpb.Property._should_validate = True
         assert bcpb.validation_on()
 
+    def test__hinted_value_is_identity(self) -> None:
+        p = bcpb.Property()
+        assert p._hinted_value(10, "hint") == 10
```

Comment:
This and later `ColumnDataChangedEvent("doc", ...)` makes those tests unrobust under future changes to code logic. Also from static typing perspective it leaves work for, most likely, me in future to rewrite this into complete code.

Reply:
I still maintain that it is not in general possible or desirable to type check  all test code. Test code needs to simulate usage to an extent, and in particular we often need to explicitly test behaviors that occur when users pass in unexpected values (i.e. calling with a *wrong type*, that mypy would complain about) to maintain that error messages are properly generated, etc. But more fundamentally, I absolutely believe that on balance it is far more important and better for the project to make tests easy to write and contribute, so that there are more tests. 

As for this specific test, I actually think the opposite of your claim is true.  A dummy value here maintains that this base class version behaves as the identity, independent of both value *and type*. That is a stronger claim, not a weaker one. You could argue that as a private method, all usage of this method fall under our mypy check, we therefore don't care about what users might do with it, and so limiting to real hints acceptable. But even then, why bother? What does that extra work and barriers to writing tests buy us for the effort? 

In my ideal scenario, we would:

* type all test functions, i.e. `test_foo`, themselves
* type all test utility functions or classes
* not worry about types for usage *of the library* inside tests, at all

But I don't think mypy will easily support this. So then the question becomes one of priorities, and not molding ourselves to a tool just for its own sake.